pkg/storegateway/storepb/custom_test.go

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package storepb
+
+import (
+	"testing"
+	"unsafe"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/mimir/pkg/mimirpb"
+)
+
+func TestSeries_CloneUnsafe(t *testing.T) {
+	const (
+		origLabelName  = "name"
+		origLabelValue = "value"
+	)
+	labelNameBytes := []byte(origLabelName)
+	labelValueBytes := []byte(origLabelValue)
+	s := Series{
+		Labels: []mimirpb.LabelAdapter{
+			{
+				Name:  yoloString(labelNameBytes),
+				Value: yoloString(labelValueBytes),
+			},
+		},
+		Chunks: []AggrChunk{
+			{
+				Raw: Chunk{
+					Data: mimirpb.UnsafeByteSlice(labelValueBytes),
```

Comment:
Same as above, reusing the same byte slice here seems confusing to me.

Reply:
I re-used because I didn't see any strict need to have a separate reference for the chunk, but I've pushed a revision with a dedicated one.